File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/Delphi/UnboundColumnsDemo/UnboundColumnsDemoData.pas
unit UnboundColumnsDemoData;

interface

uses
  Forms, SysUtils, Classes, DB, cxStyles, ImgList, Controls, cxEdit,
  cxEditRepositoryItems, cxDBEditRepository, cxClasses, cxGridCardView, DBClient, MidasLib;

type
  TUnboundColumnsDemoDataDM = class(TDataModule)
    tblCustomers: TClientDataSet;
    dsCustomers: TDataSource;
    tblCustomersID: TAutoIncField;
    tblCustomersFIRSTNAME: TStringField;
    tblCustomersLASTNAME: TStringField;
    tblCustomersCOMPANYNAME: TStringField;
    tblCustomersPAYMENTTYPE: TIntegerField;
    tblCustomersPRODUCTID: TIntegerField;
    tblCustomersCUSTOMER: TBooleanField;
    tblCustomersPURCHASEDATE: TDateTimeField;
    tblCustomersPAYMENTAMOUNT: TCurrencyField;
    tblCustomersCOPIES: TIntegerField;
    procedure DataModuleCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  UnboundColumnsDemoDataDM: TUnboundColumnsDemoDataDM;

implementation

{$R *.dfm}

procedure TUnboundColumnsDemoDataDM.DataModuleCreate(Sender: TObject);
begin
  tblCustomers.LoadFromFile(ExtractFilePath(Application.ExeName) + '..\..\Data\Customer.xml');
end;

end.

File: Components/DevExpressVCL/Demos/ExpressQuantumGrid/CBuilder/UnboundColumnsDemo/UnboundColumnsDemoData.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "UnboundColumnsDemoData.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "cxClasses"
#pragma link "cxGridTableView"
#pragma link "cxStyles"
#pragma link "cxDBEditRepository"
#pragma link "cxEdit"
#pragma link "cxEditRepositoryItems"
#pragma link "cxGridCardView"
#pragma resource "*.dfm"
TUnboundColumnsDemoDataDM *UnboundColumnsDemoDataDM;
//---------------------------------------------------------------------------
__fastcall TUnboundColumnsDemoDataDM::TUnboundColumnsDemoDataDM(TComponent* Owner)
  : TDataModule(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TUnboundColumnsDemoDataDM::DataModuleCreate(TObject *Sender)
{
    tblCustomers->LoadFromFile(ExtractFilePath(Application->ExeName) + "..\\..\\Data\\Customer.xml");
};
//---------------------------------------------------------------------------
